lib/appsignal/hooks/shoryuken.rb

```diff
@@ -1,21 +1,19 @@
 module Appsignal
   class Hooks
     class ShoryukenMiddleware
-      def call(_worker_instance, queue, sqs_msg, body)
-        metadata = {
-          :queue => queue
-        }
-        exclude_keys = [:job_class, :queue_name, :arguments]
-        metadata.merge!(body.reject { |key| exclude_keys.member?(key.to_sym) })
-        metadata.merge!(sqs_msg.attributes)
-
+      def call(worker_instance, queue, sqs_msg, body)
+        metadata = { :queue => queue }.merge(sqs_msg.attributes)
         options = {
-          :class => body["job_class"],
+          :class => worker_instance.class.name,
           :method => "perform",
           :metadata => metadata
         }
-        options[:params] = body["arguments"] if body.key?("arguments")
-        options[:queue_start] = Time.at(sqs_msg.attributes["SentTimestamp"].to_i / 1000) if sqs_msg.attributes.key?("SentTimestamp")
+
+        options[:params] = body.is_a?(Hash) ? body : { :params => body }
```

Comment:
Can we wrap the params with a `format_args` call? This truncates the arguments if they are to long, (e.g. if one of the params is a 10mb XML file, we would prefer not to store it on AppSignal :) 

This is how we use it for Sidekiq: https://github.com/appsignal/appsignal-ruby/blob/master/lib/appsignal/hooks/sidekiq.rb#L16